File: python/数组/1720. 解码异或后的数组.py
from typing import List
class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        i = 0
        res = [first]
        before = first
        while i<len(encoded):
            t = before^encoded[i]
            res.append(t)
            before = t
            i+=1
        return res

File: go/array/1720. 解码异或后的数组.go
package array

func decode(encoded []int, first int) []int {
	before := first
	var res []int
	res = append(res, first)
	for i := 0; i < len(encoded); i++ {
		t := before ^ encoded[i]
		res = append(res, t)
		before = t
	}
	return res
}
